### PR TITLE
gas optimization

### PR DESCRIPTION
**Pull request by lirona**
avg gas before changes: 6937807
after 1st commit: 6908110
after 2nd commit: 6814081
after 3rd commit: 6805686

### CODE REVIEW

File: contracts/HATVault.sol [hunk 1]
```diff
@@ -189,35 +189,36 @@ contract HATVault is IHATVault, ERC4626Upgradeable, OwnableUpgradeable, Reentran
...
-        claimId = keccak256(abi.encodePacked(address(this), nonce++));
+        claimId = keccak256(abi.encodePacked(address(this), ++nonce));
         activeClaim = Claim({
             claimId: claimId,
             beneficiary: _beneficiary,
             bountyPercentage: _bountyPercentage,
             committee: msg.sender,
             // solhint-disable-next-line not-rely-on-time
-            createdAt: block.timestamp,
+            createdAt: uint32(block.timestamp),
```
Comment (ben-kaufman) [inline contracts/HATVault.sol:221]: So this will work for about 83 years. God enough I guess :)

File: contracts/HATVault.sol [hunk 2]
```diff
@@ -237,23 +238,21 @@ contract HATVault is IHATVault, ERC4626Upgradeable, OwnableUpgradeable, Reentran
     }
 
     function challengeClaim(bytes32 _claimId) external isActiveClaim(_claimId) {
-        if (
-            registry.owner() != msg.sender &&
-            activeClaim.arbitrator != msg.sender
-        ) revert OnlyArbitratorOrRegistryOwner();
+        if (msg.sender !=  activeClaim.arbitrator && msg.sender != registry.owner())
```
Comment (ben-kaufman) [inline contracts/HATVault.sol:241]: Double space

File: contracts/HATVaultsRegistry.sol [hunk 3]
```diff
@@ -336,22 +341,27 @@ contract HATVaultsRegistry is IHATVaultsRegistry, Ownable {
...
+            unchecked { ++i; }
         }
         if (swapData.amount == 0) revert AmountToSwapIsZero();
         IERC20 _HAT = HAT;
         (swapData.hatsReceived, swapData.amountUnused) = _swapTokenForHAT(IERC20(_asset), swapData.amount, _amountOutMinimum, _routingContract, _routingPayload);
         
-        swapData.governanceAmountSwapped = (swapData.amount - swapData.amountUnused).mulDiv(governanceHatReward[_asset], swapData.amount);
-        governanceHatReward[_asset] = swapData.amountUnused.mulDiv(governanceHatReward[_asset], swapData.amount);
+        swapData.usedPart = (swapData.amount - swapData.amountUnused) / swapData.amount;
+        swapData.governanceAmountSwapped = swapData.usedPart * swapData.governanceHatReward;
```
Comment (ben-kaufman) [inline contracts/HATVaultsRegistry.sol:357]: Hmm this gives up the mulDiv use.

File: contracts/HATVaultsRegistry.sol [hunk 4]
```diff
@@ -336,22 +341,27 @@ contract HATVaultsRegistry is IHATVaultsRegistry, Ownable {
...
+        governanceHatReward[_asset]  = swapData.amountUnused.mulDiv(swapData.governanceHatReward, swapData.amount);
 
-        for (uint256 i = 0; i < _beneficiaries.length; i++) {
-            uint256 hackerReward = swapData.hatsReceived.mulDiv(hackersHatReward[_asset][_beneficiaries[i]], swapData.amount);
-            uint256 hackerAmountSwapped = (swapData.amount - swapData.amountUnused).mulDiv(hackersHatReward[_asset][_beneficiaries[i]], swapData.amount);
+        for (uint256 i = 0; i < _beneficiaries.length;) {
+            uint256 hackerReward = swapData.hatsReceived.mulDiv(swapData.hackerRewards[i], swapData.amount);
+            uint256 hackerAmountSwapped = swapData.usedPart * swapData.hackerRewards[i];
```
Comment (ben-kaufman) [inline contracts/HATVaultsRegistry.sol:362]: also this gives up the mulDiv use.

File: contracts/HATVaultsRegistry.sol [hunk 5]
```diff
@@ -373,15 +383,37 @@ contract HATVaultsRegistry is IHATVaultsRegistry, Ownable {
                 _HAT.safeTransfer(tokenLock, hackerReward);
             }
             emit SwapAndSend(_beneficiaries[i], hackerAmountSwapped, hackerReward, tokenLock);
+            unchecked { ++i; }
         }
-        _HAT.safeTransfer(owner(), swapData.hatsReceived - swapData.totalHackerReward);
-        emit SwapAndSend(owner(), swapData.governanceAmountSwapped, swapData.hatsReceived - swapData.totalHackerReward, address(0));
+        address _owner = owner(); 
+        uint256 _amountToOwner= swapData.hatsReceived - swapData.totalHackerReward;
```
Comment (ben-kaufman) [inline contracts/HATVaultsRegistry.sol:389]: Space missing

File: contracts/RewardController.sol [hunk 6]
```diff
@@ -103,26 +118,29 @@ contract RewardController is IRewardController, OwnableUpgradeable {
...
-        }  else {
-            uint256 nextEpoch = (block.number - startBlock) / epochLength + 1;
+            emit SetEpochRewardPerBlock(_epochRewardPerBlock);
+        } else {
+            uint256 nextEpoch = (block.number - _startBlock) / epochLength + 1;
             // if rewards are ongoing, update the future rewards but keep past and current
-            for (; nextEpoch < NUMBER_OF_EPOCHS; nextEpoch++) {
-                epochRewardPerBlock[nextEpoch] = _epochRewardPerBlock[nextEpoch];
+            for (; nextEpoch < NUMBER_OF_EPOCHS; ++nextEpoch) {
+                epochRewardPerBlock[nextEpoch] = _epochRewardPerBlock[nextEpoch];
```
Comment (ben-kaufman) [inline contracts/RewardController.sol:140]: Remove space at the end

File: contracts/HATVaultsRegistry.sol (hunk 3 above)
Comment (lirona) [reply contracts/HATVaultsRegistry.sol:357]: hmm right.. maybe better to have it with mulDiv actually

**Review by jellegerbrandy — APPROVED** [review]

**Pull request merged**